plugins/modem-manager/fu-mm-device.c

```diff
@@ -1279,10 +1217,106 @@ fu_mm_device_write_firmware_mbim_qdu(FuDevice *device,
 #endif /* MM_CHECK_VERSION(1,17,1) && MBIM_CHECK_VERSION(1,25,3) */
 
 #if MM_CHECK_VERSION(1, 17, 2)
+static gboolean
+fu_mm_device_find_edl_port(FuDevice *device, const gchar *subsystem, GError **error)
+{
+	FuMmDevice *self = FU_MM_DEVICE(device);
+
+	g_clear_pointer(&self->port_edl, g_free);
+
+	for (guint i = 0; i < 30; i++) {
+		if (fu_mm_utils_find_device_file(fu_device_get_physical_id(FU_DEVICE(self)),
+						 subsystem,
+						 &self->port_edl,
+						 NULL))
+			return TRUE;
+
+		g_usleep(250 * 1000);
+	}
```

Comment:
could you use fu_device_retry_full() here? I think it would be cleaner.

Reply:
yep, changed to fu_device_retry_full()